src/promptflow/tests/sdk_cli_azure_test/conftest.py

```diff
@@ -186,19 +195,34 @@ def mock_set_headers_with_user_aml_token(mocker: MockerFixture) -> None:
     There will be requests fetching cloud metadata during retrieving AML token, which will break during replay.
     As the logic comes from azure-ai-ml, changes in Prompt Flow can hardly affect it, mock it here.
     """
-    mocker.patch("promptflow.azure._restclient.flow_service_caller.FlowServiceCaller._set_headers_with_user_aml_token")
-    return
+    if not is_live():
```

Comment:
I see `is_live` and `is_replay`, are they actually the opposite? if yes maybe we don't need both of them?

Reply:
currently we have three modes: live, record and replay, so maybe we cannot simply remove any of them